hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -253,7 +265,16 @@ public TimelineLayoutVersion getTimelineLayoutVersion() {
    */
   public HoodieWrapperFileSystem getFs() {
     if (fs == null) {
-      FileSystem fileSystem = FSUtils.getFs(metaPath, hadoopConf.newCopy());
+      FileSystem fileSystem;
+      if (fileSystemRetryConfig.isFileSystemActionRetryEnable()) {
```

Comment:
could we make this a property of the `StorageScheme`?

Reply:
Emmm..., it's easy to make this as a property of the `StorageScheme` something like `FILE("file", append: false, retry: true)`
What I am more worried about is that this may mean that the retry function is turned on by default. And it may be hard for users to disable this retry if they don't need it.

Reply:
we can disable by default and let interested users enable on their end.

Reply:
Guess, what vinoth refers to is, we should add a property named supportsRetry or something to each StorageScheme. So, only for those storage schemes where retries are supported, we can check the config (fileSystemRetryConfig.isFileSystemActionRetryEnable()) and then invoke our retry Helper. 

@vinothchandar : can you please clarify on your above suggestion.

Reply:
In that case, I think we can just have the config alone. and let the user decide.

Reply:
Ack and changed as your suggestion. Thanks a lot.